### ISSUE TITLE
Optimal Scaling

### ISSUE THREAD

**Issue by ofivite** [Helmholtz-AI-Energy/HAICON25-Prologue-Day#16]
### _If you need some help, check out these useful [tips](https://www.ctl.ox.ac.uk/planning-effective-tutorials) for organizing tutorials!_
---

# Title
Optimal scaling

# Description
This tutorial will focus on optimal scaling in machine learning—a principled approach to training models efficiently and effectively by leveraging scaling laws, optimal hyperparameter transfer, and compute-efficient training design. Participants will be guided through both theoretical foundations and hands-on sessions to understand how to scale model training with the least required compute resources. 

The theoretical part of the tutorial will include an overview from recent papers introducing the fundamental methodology and corresponding terminology such as zero-shot hyperparameter transfer, model parametrisation, critical batch size and steepest descend under modular norm:

- [Tensor Programs V: Tuning Large Neural Networks via Zero-Shot Hyperparameter Transfer](https://arxiv.org/abs/2203.03466)
- [Time Transfer: On Optimal Learning Rate and Batch Size In The Infinite Data Limit](https://arxiv.org/abs/2410.05838)
- [Modular Duality in Deep Learning](https://arxiv.org/abs/2410.21265)
- [Training Deep Learning Models with Norm-Constrained LMOs](https://arxiv.org/abs/2502.07529)

The hands-on part will introduce participants to the usual workflow of scaling up the model and dataset size, using a simple transformer model on text data to perform HP tuning on the small scale and observe the model dynamics and hyperparameter evolution in the model and data size scaling limits.
 
## Goals
- Equip participants with a rigorous understanding of how model and dataset scaling affects performance and efficiency.
- Deepen the understanding of model dynamics evolution in the scaling limits.
- Teach practical skills in designing the model architecture, selecting compute-optimal hyperparameters, and using scaling laws.
- Bridge the gap between cutting-edge research and practical implementation.

This tutorial is highly relevant for HAICON 2025 as it directly addresses the growing need for scalable, cost-effective ML methods in resource-constrained environments of the academic world. As model sizes and datasets continue to grow, understanding _how_ to scale is as crucial as _what_ to scale.

# Organizational

## Host(s)
Oleg Filatov ([o.filatov@fz-juelich.de](mailto:o.filatov@fz-juelich.de)) @ofivite
Jiangtao Wang ([jian.wang@fz-juelich.de](mailto:jian.wang@fz-juelich.de)) @rakkit
Jan Ebert  ([ja.ebert@fz-juelich.de](mailto:ja.ebert@fz-juelich.de)) @janEbert

## Format
- 90 min: theory lecture + discussions (part 1, morning)
- 90 min: hands-on exercises + discussions (part 2, afternoon)

## Number of participants
10-20

## Material
Projector
Power outlets
Tables

## Pre-Requisites
For hands-on:
- Laptop (ideally participants bring their own, but not super required)
- Knowledge of Python, PyTorch, ML fundamentals, slurm, and bash
- We will have compute hours on the JURECA supercomputer available for the course participants

## Preferred Slot *(Morning, Afternoon or No preference)*
Morning + Afternoon

## Want to host the tutorial session in both slots?
Yes, we will have a theoretical introduction in the morning and practical hands-on in the afternoon

## What's next?
Thanks for creating this proposal! For now, you don't have to do anything else. Once we have received all the proposals and are developing the final prologue day program we will be in touch again to confirm any final details (time slot, room, material etc.). If you have any questions please don't hesitate to reach out [HAI Consultant Team - KIT](mailto:consultant-helmholtz.ai@kit.edu).

**Comment by kalebphipps** [comment]
Hi,

everything is set from our side for next week. Do you require any more assistance, e.g sending information out to all registered participants or similar?

Look forward to next week!
